datashader/utils.py

```diff
@@ -391,7 +393,14 @@ def dshape_from_pandas(df):
 
 def dshape_from_dask(df):
     """Return a datashape.DataShape object given a dask dataframe."""
-    return datashape.var * dshape_from_pandas(df.head()).measure
+    cat_columns = [col for col in df.columns if isinstance(df[col].dtype, type(pd.Categorical.dtype))
+                   or isinstance(df[col].dtype, pd.api.types.CategoricalDtype)]
+    if len(cat_columns) > 1:
+        # If there is more than one categorical column it is faster
+        # to compute the df.head() which will contain all categories
+        return datashape.var * dshape_from_pandas(df.head()).measure
```

Comment:
I don't suppose it will get any faster if you specify `df.head(1)`, but maybe it's more explicit?  No strong opinion.

Reply:
I tried, it's slower.

Reply:
Hmm; that's very strange.  Is `df.head(5)` (or whatever the default is) _also_ slower?  Seems fishy!

Reply:
Not too surprising, it still has load the first partition and the network overhead difference between 1 and 5 rows is tiny. So the only difference is that it adds another task to the graph which runs the iloc.